numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -1887,6 +1887,18 @@ static PyObject *
 }
 /**end repeat**/
 
+static PyObject *
+voidtype_keys(PyObject *self, PyObject *NPY_UNUSED(args))
+{
```

Comment:
This function can now be called with either a `PyArrayObject*` or a `PyVoidArrObject`, depending on how it is accessed.

Perhaps this is too dangerous, and instead both objects should defer to some external dispatch table